core/templates/dev/head/pages/exploration_player/ProgressNavDirective.js

```diff
@@ -58,8 +58,9 @@ oppia.directive('progressNav', [
                 $scope.activeCard.stateName));
             $scope.interactionCustomizationArgs = interaction.customizationArgs;
             $scope.interactionId = interaction.id;
-            interactionHasNavSubmitButton = INTERACTION_SPECS[
-              interaction.id].show_nav_submit_button;
+            interactionHasNavSubmitButton = (
+              Boolean(interaction.id) &&
```

Comment:
Out of curiosity, this is because the id might not exist in solution editor?

Reply:
Oh no. This is for the editor view. If you create a new exploration and then preview it immediately, there's no interaction specified yet. This was causing a failure in one of the e2e tests.

Reply:
Ah OK I see, thanks!